src/Workspaces/CSharp/Portable/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -572,6 +565,44 @@ private static bool AtEndOfIncompleteStringOrCharLiteral(SyntaxToken token, int
             {
                 selectedMembers.Free();
             }
+
+            bool IsSelectedFieldOrProperty(MemberDeclarationSyntax member)
+            {
+                if (!member.IsKind(SyntaxKind.FieldDeclaration, SyntaxKind.PropertyDeclaration))
+                {
+                    return false;
+                }
+
+                // first, check if entire member is selected
+                if (textSpan.Contains(member.Span))
+                {
+                    return true;
+                }
+
+                if (!allowPartialSelection)
+                {
+                    return false;
+                }
+
+                // next, check if identifier is at least partially selected
+                switch (member.Kind())
```

Comment:
just do a normal `switch (member)`  and do `case FieldDeclarationSyntax field: ...`

Reply:
Nice!  fixed it.